### PR TITLE
Add VM Extension Operator

### PR DESCRIPTION
**Pull request by huangpf**
**What this PR does / why we need it**:
This PR adds a VM extension operator for one of the crucial Compute use cases.

**Special notes for your reviewer**:

**How does this PR make you feel**:
^-^

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -47,7 +47,7 @@ RUN apt-get update \
         golang.org/x/lint/golint \
         github.com/alecthomas/gometalinter \
         honnef.co/go/tools/... \
-        github.com/golangci/golangci-lint/cmd/golangci-lint \
+        #github.com/golangci/golangci-lint/cmd/golangci-lint \
```
Comment (frodopwns) [inline .devcontainer/Dockerfile:50]: what was the issue here? doesn't work anymore?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -93,7 +93,7 @@ COPY ./Makefile ./
 RUN make install-kind
 RUN make install-kubebuilder
 RUN make install-kustomize
-RUN make install-test-dependency
+# RUN make install-test-dependency
```
Comment (frodopwns) [inline .devcontainer/Dockerfile:96]: is this change widely applicable or was it a local workaround?

File: config/default/manager_image_patch.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: controller:latest
+      - image: IMAGE_URL
```
Comment (frodopwns) [inline config/default/manager_image_patch.yaml:11]: will need to undo this change

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (huangpf) [reply .devcontainer/Dockerfile:96]: I think Janani/Justin confirmed it this target we removed/renamed. Maybe I should just remove it.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (huangpf) [reply .devcontainer/Dockerfile:50]: I think it's a dynamic package getting downloaded into the container, and it started to error out recently. So, I had to disable it to make the container build process work. Justin sent out this full fix PR: https://github.com/Azure/azure-service-operator/pull/1059

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (huangpf) [reply .devcontainer/Dockerfile:96]: I will try to change it to ```install-test-dependencies``` as Janani indicated.

**Comment by huangpf** [comment]
Closing this one, and moving it to a newly created one: https://github.com/Azure/azure-service-operator/pull/1087
